src/main/java/net/snowflake/client/core/SFBaseResultSet.java

```diff
@@ -284,7 +283,10 @@ protected SfSqlArray getJsonArray(String obj, int columnIndex) throws SFExceptio
       int columnType = ColumnTypeHelper.getColumnType(columnSubType, session);
       int scale = fieldMetadata.getScale();
 
-      ArrayNode arrayNode = (ArrayNode) OBJECT_MAPPER.readTree(obj);
+      // Nowadays the null value inside arrays is returned as 'undefined' what is incompatible with json format.
+      // Must be change to null
+      String fixedArrayObject = obj.replaceAll("undefined", "null");
```

Comment:
I don't believe this is good... Let's suppose we have `["something is undefined!"]`, what happens then?

Reply:
Yep It is more complicated